source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,3065 @@
+  monitoring *and* SDAM events) for a single client. See
+  `Mixing event types in observeEvents and expectEvents`_ for more
+  information.
+
+.. _test_outcome:
+
+- ``outcome``: Optional array of one or more `collectionData`_ objects. Data
+  that is expected to exist in collections after each test case is executed.
+
+  The list of documents herein SHOULD be sorted ascendingly by the ``_id`` field
+  to allow for deterministic comparisons. The procedure for asserting collection
+  contents is discussed in `Executing a Test`_.
+
+
+operation
+~~~~~~~~~
+
+An operation to be executed as part of the test.
+
+The structure of this object is as follows:
+
+.. _operation_name:
+
+- ``name``: Required string. Name of the operation (e.g. method) to perform on
+  the object.
+
+.. _operation_object:
+
+- ``object``: Required string. Name of the object on which to perform the
+  operation. This SHOULD correspond to either an `entity`_ name (for
+  `Entity Test Operations`_) or "testRunner" (for `Special Test Operations`_).
+  If the object is an entity, The YAML file SHOULD use an `alias node`_ for its
+  ``id`` field (e.g. ``object: *collection0``).
+
+.. _operation_arguments:
+
+- ``arguments``: Optional object. Map of parameter names and values for the
+  operation. The structure of this object will vary based on the operation.
+  See `Entity Test Operations`_ and `Special Test Operations`_.
+
+  The ``session`` parameter is handled specially (see `commonOptions_session`_).
+
+.. _operation_expectError:
+
+- ``expectError``: Optional `expectedError`_ object. One or more assertions for
+  an error expected to be raised by the operation.
+
+  This field is mutually exclusive with
+  `expectResult <operation_expectResult_>`_ and
+  `saveResultAsEntity <operation_saveResultAsEntity_>`_.
+
+  This field SHOULD NOT be used for `Special Test Operations`_ (i.e.
+  ``object: testRunner``).
+
+.. _operation_expectResult:
+
+- ``expectResult``: Optional mixed type. A value corresponding to the expected
+  result of the operation. This field may be a scalar value, a single document,
+  or an array of values. Test runners MUST follow the rules in
+  `Evaluating Matches`_ when processing this assertion.
+
+  This field is mutually exclusive with `expectError <operation_expectError_>`_.
+
+  This field SHOULD NOT be used for `Special Test Operations`_ (i.e.
+  ``object: testRunner``).
+
+.. _operation_saveResultAsEntity:
+
+- ``saveResultAsEntity``: Optional string. If specified, the actual result
+  returned by the operation (if any) will be saved with this name in the
+  `Entity Map`_.  The test runner MUST raise an error if the name is already in
+  use or if the result does not comply with `Supported Entity Types`_.
+
+  This field is mutually exclusive with `expectError <operation_expectError_>`_.
+
+  This field SHOULD NOT be used for `Special Test Operations`_ (i.e.
+  ``object: testRunner``).
+
+
+expectedError
```

Comment:
Should this be `expectError` ?

Reply:
The directives in the test format are active phrases like `expectError`, but the structs used internally in the spec document are subjects, like `expectedError`. I also did this for the event structs with `expectEvents` and `expectedEventsForClient`.